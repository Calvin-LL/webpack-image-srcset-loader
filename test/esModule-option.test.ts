import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe('"esModule" option', () => {
  test("should work with default true", async () => {
    const compiler = getCompiler(
      {
        sizes: [],
      },
      false,
      "simple-require.js"
    );
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  test("should work with true", async () => {
    const compiler = getCompiler(
      {
        sizes: [],
        esModule: true,
      },
      false,
      "simple-require.js"
    );
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  test("should work with false", async () => {
    const compiler = getCompiler(
      {
        sizes: [],
        esModule: false,
      },
      false,
      "simple-require.js"
    );
    const stats = await compile(compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });
});
