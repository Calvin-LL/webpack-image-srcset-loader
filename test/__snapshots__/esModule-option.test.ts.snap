// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"esModule" option should work with default true: result 1`] = `
Object {
  "default": "",
}
`;

exports[`"esModule" option should work with false: result 1`] = `""`;

exports[`"esModule" option should work with true: result 1`] = `
Object {
  "default": "",
}
`;
